Panic when sending time.Time

Somewhere in this library will panic when sending a golang `time.Time`:

```
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method
```

The golang reflect documentation suggests to use `reflect.Value.CanInterface()` before calling `reflect.Value.Interface()`. If !v.CanInterface(), we could use `reflect.Value.String()` instead or the unexported access technique in the below link.

- https://forum.golangbridge.org/t/panic-reflect-value-interface-cannot-return-value-obtained-from-unexported-field-or-method/17462

- https://stackoverflow.com/questions/42664837/access-unexported-fields-in-golang-reflect/43918797#43918797

- https://play.golang.org/p/IgjlQPYdKFR

--- a/etf/encode.go
+++ b/etf/encode.go
@@ -23,7 +23,7 @@ func Encode(term Term, b *lib.Buffer,
 	writerAtomCache map[Atom]CacheItem,
 	encodingAtomCache *ListAtomCache) (retErr error) {
 	defer func() {
-		// We should catch any panic happend during encoding the raw data.
+		// We should catch any panic happend during encoding Golang types.
 		if r := recover(); r != nil {
 			retErr = fmt.Errorf("%v", r)
 		}

--- a/etf/encode.go
+++ b/etf/encode.go
@@ -3,10 +3,11 @@ package etf
 import (
 	"encoding/binary"
 	"fmt"
-	"github.com/halturin/ergo/lib"
 	"math"
 	"math/big"
 	"reflect"
+
+	"github.com/halturin/ergo/lib"
 )
 
 var (
@@ -20,7 +21,13 @@ var (
 func Encode(term Term, b *lib.Buffer,
 	linkAtomCache *AtomCache,
 	writerAtomCache map[Atom]CacheItem,
-	encodingAtomCache *ListAtomCache) error {
+	encodingAtomCache *ListAtomCache) (retErr error) {
+	defer func() {
+		// We should catch any panic happend during encoding the raw data.
+		if r := recover(); r != nil {
+			retErr = fmt.Errorf("%v", r)
+		}
+	}()
 
 	var stack, child *stackElement
 

